local-cli/automated-vs/solutions/vsSolution.js

```diff
@@ -0,0 +1,142 @@
+ /*
+  * Some descendant of this class needs to have:
+  * deploy
+  */
+
+'use-strict';
+
+const execSync = require('child_process').execSync;
+const path = require('path');
+const chalk = require('chalk');
+const fs = require('fs');
+const shell = require('shelljs');
+const child_process = require('child_process');
+const EOL = require('os').EOL;
+const nuget = require('../../nuget-exe/index');
+const MSBuild = require('../utils/msbuild');
+const MSBuildTools = MSBuild.MSBuildTools;
+const VS2017 = require('../visualStudios/vs2017');
+const VS2015OrEarlier = require('../visualStudios/vs2015OrEarlier');
+const utils = require('../utils/utility');
+
+module.exports = class VSSolution {
+    constructor(slnFile, root) {
+        this.slnFile = slnFile;
+        this.root = root;
+        this.name = this.getName(slnFile);
+        this.visualStudio = this.createVSVersion();
```

Comment:
The Visual Studio object should be injected into the constructor (aka passed in as an object).

Reply:
👍